### DISCUSSION TITLE
Version-Tagged Docker Images for Better Integration and Automation

### DISCUSSION THREAD

**Discussion by kersten** [inovex/scrumlr.io#4078] [Proposals]
Hello community,

I am currently exploring the deployment of Scrumlr.io in our Kubernetes cluster and have noticed that the Docker images available are only tagged with 'latest' or specific commit references. This setup presents some challenges in maintaining a stable and manageable deployment, especially in a production environment.

## Proposal:
I propose that Docker images for each release of Scrumlr.io be tagged with the corresponding version number in addition to the existing tagging strategy. This change would enable several improvements in the deployment process:

**Stability and Traceability:** Version-tagged images ensure that teams can deploy specific, known versions of Scrumlr.io, which aids in debugging and tracking software versions across multiple environments.

**Integration with CI/CD Pipelines:** Having images tagged by version number allows automated pipelines to easily reference and pull the specific versions needed without manual intervention, streamlining the update process.

**Automation Tool Compatibility:** Tools like Renovate and Dependabot, which are widely used for dependency updates, require version-specific tags to function correctly. They could automatically propose updates when a new version of Scrumlr.io is released, enhancing the security and currency of deployments.

Implementing this would not only benefit those of us looking to integrate Scrumlr.io more deeply into automated environments but also improve the overall management and operability for everyone using the platform in different contexts.

Thank you for considering this proposal. I believe these changes would greatly benefit the community by facilitating easier, safer, and more consistent deployments of Scrumlr.io.

Best regards

Kersten

**Comment by Lennart01** [comment]
All images are already tagged on release. 
https://github.com/inovex/scrumlr.io/actions?query=event%3Arelease
`image:version`
eg
`:3.7.1`
Maybe we could document this somewhere 🤔.

**Reply by kersten** [reply]
Ok, sorry. I missed that :) Thanks for the hint. The reason I encountered this issue was because I didn't read the details thoroughly enough :) My Kubernetes deployment failed with the following error, indicating that it could not find the images for my platform:

```
Failed to pull image "ghcr.io/inovex/scrumlr.io/scrumlr-frontend:3.7.1": rpc error: code = NotFound desc = failed to pull and unpack image "ghcr.io/inovex/scrumlr.io/scrumlr-frontend:3.7.1": no match for platform in manifest: not found
```

This issue is discussed in more detail here: https://github.com/inovex/scrumlr.io/discussions/3994

**Reply by Lennart01** [reply]
You can try ghcr.io/inovex/scrumlr.io/scrumlr-server:sha-bcb3adc@sha256:f0d7c3cb09f17631394f8b5d9adc3ceed387b0fbd7f41ec8656fe09262a2f597
This is an arm image we built during testing
